stdlib/contextlib.pyi

```diff
@@ -113,7 +113,8 @@ if sys.version_info >= (3, 7):
         ) -> Awaitable[bool]: ...
 
 if sys.version_info >= (3, 7):
-    @overload
-    def nullcontext(enter_result: _T) -> ContextManager[_T]: ...
-    @overload
-    def nullcontext() -> ContextManager[None]: ...
+    class nullcontext(AbstractContextManager[_T]):
+        enter_result: _T
+        def __init__(self, enter_result: _T=...) -> None: ...
```

Comment:
I think this still needs overloads. Otherwise mypy won't know that `_T` should be `None` for `nullcontext()` without any arguments. Something like this:

```suggestion
        @overload
        def __init__(self: nullcontext[None], enter_result: None = ...) -> None: ...
        @overload
        def __init__(self: nullcontext[_T], enter_result: _T) -> None: ...
```